examples/rules.dl

```diff
@@ -0,0 +1,48 @@
+// Is this symbol an access path to some data?
+.decl isAccessPath(x: symbol)
+// Is this symbol a label.
+.decl isLabel(x: symbol)
+// A data flow edge
+.decl edge(src: symbol, tgt: symbol)
+ // A data flow path
+.decl path(src: symbol, tgt: symbol)
+// Definitely has this label.
+.decl hasLabel(v: symbol, label: symbol)
```

Comment:
From what I understand, it seems that we are working with two (maybe three) different concepts each currently represented with the type `symbol`:

1. Vertices (such as "P1.output", "Egress.input", etc)
2. Labels (such as "tag1", "tag2", etc)
3. (Possibly) Symbols (not to be confused with Soufle's primitive `symbol`) which is Vertex | Label

If the above is correct, we might save ourselves some confusion as this file grows if we make some base types:

```
.type Vertex <: symbol
.type Label <: symbol
.type Symbol = Vertex | Label
```

and use them instead of `symbol` in the appropriate places.